iOS_SDK/OneSignalSDK/Source/OneSignalLocation.m

```diff
@@ -186,15 +187,17 @@ + (void) internalGetLocation:(bool)prompt {
         //Location Always requires: Location Background Mode + NSLocationAlwaysUsageDescription
         NSArray* backgroundModes = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"UIBackgroundModes"];
         NSString* alwaysDescription = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"NSLocationAlwaysUsageDescription"] ?: [[NSBundle mainBundle] objectForInfoDictionaryKey:@"NSLocationAlwaysAndWhenInUseUsageDescription"];
-        if(backgroundModes && [backgroundModes containsObject:@"location"] && alwaysDescription) {
+        // use background location updates if always permission granted or prompt allowed
+        if(backgroundModes && [backgroundModes containsObject:@"location"] && alwaysDescription && (permissionStatus == 3 || prompt)) {
             [locationManager performSelector:@selector(requestAlwaysAuthorization)];
```

Comment:
I think if we move `requestWhenInUseAuthorization` and `requestAlwaysAuthorization` into their own methods and do the required checks to prevent prompting when it isn't enable might be the better approach. We can do this refracting after we merge your PR but feel free to make make larger changes if you are feeling ambitious 😄

Reply:
I just want to fix the issue.